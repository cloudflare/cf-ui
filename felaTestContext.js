// this is a wrapper for testing, it passes fela based
// components fela's renderer and global theme

import React from 'react';
import { Provider, ThemeProvider } from 'react-fela';
import { createRenderer } from './felaRenderer';
import { variables } from 'cf-style-const';

const renderer = createRenderer({ useMonolithic: true });

export default component => (
  <Provider renderer={renderer}>
    <ThemeProvider theme={variables}>{component}</ThemeProvider>
  </Provider>
);
