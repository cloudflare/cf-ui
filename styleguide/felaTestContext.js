import React from 'react';
import { Provider } from 'react-fela';
import { createRenderer } from './felaRenderer';

const renderer = createRenderer({ useMonolithic: true });

export default component => (
  <Provider renderer={renderer}>{component}</Provider>
);
