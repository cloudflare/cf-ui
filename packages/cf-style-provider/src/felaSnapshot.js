import renderer from 'react-test-renderer';
import { html as beautify } from 'js-beautify';
import CleanCSS from 'clean-css';
import felaTestContext from './felaTestContext';
import createRenderer from './createRenderer';

const cssCleaner = new CleanCSS({ format: 'beautify', level: 0 });

export default component => {
  const felaRenderer = createRenderer({
    dev: true
  });

  return {
    component: renderer
      .create(felaTestContext(component, felaRenderer))
      .toJSON(),
    styles: cssCleaner.minify(felaRenderer.renderToString()).styles
  };
};
