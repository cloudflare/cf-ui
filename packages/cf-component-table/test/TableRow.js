import React from 'react';
import renderer from 'react-test-renderer';
import {
  TableRow,
  TableCell,
  createTableRow
} from '../../cf-component-table/src/index';
import felaTestContext from '../../cf-style-provider/src/felaTestContext';
import felaSnapshot from '../../cf-style-provider/src/felaSnapshot';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('TableRow', () => {
  it('should render', () => {
    const snapshot = felaSnapshot(<TableRow>TableRow</TableRow>);
    expect(snapshot).toMatchSnapshot();
  });

  it('should render with type', () => {
    const snapshot = felaSnapshot(<TableRow type="error">TableRow</TableRow>);
    expect(snapshot).toMatchSnapshot();
  });

  it('should pass down CSS query props', () => {
    const wrapper = mount(
      felaTestContext(
        <table>
          <tbody>
            <TableRow
              condensed
              striped
              bordered
              hover
              bare
              type="info"
              theadIndex={0}
              tbodyIndex={0}
              tfootIndex={0}
              rowIndex={0}
            >
              <TableCell />
              <TableCell />
            </TableRow>
          </tbody>
        </table>
      )
    );

    const cells = wrapper.find(TableCell);

    for (let i = 0; i < 2; i++) {
      expect(cells.at(i).prop('condensed')).toBeTruthy();
      expect(cells.at(i).prop('striped')).toBeTruthy();
      expect(cells.at(i).prop('bordered')).toBeTruthy();
      expect(cells.at(i).prop('hover')).toBeTruthy();
      expect(cells.at(i).prop('bare')).toBeTruthy();
      expect(cells.at(i).prop('rowType')).toBe('info');
      expect(cells.at(i).prop('theadIndex')).toBe(0);
      expect(cells.at(i).prop('tbodyIndex')).toBe(0);
      expect(cells.at(i).prop('tfootIndex')).toBe(0);
      expect(cells.at(i).prop('rowIndex')).toBe(0);
    }

    expect(cells.at(0).prop('cellIndex')).toBe(0);
    expect(cells.at(1).prop('cellIndex')).toBe(-1);
  });

  it('should compose with styles overrides', () => {
    const TableRow = createTableRow(({ theme }) => ({
      verticalAlign: 'right',
      [`@media (min-width: ${theme.breakpoints.desktopLarge})`]: {
        width: '1000px'
      }
    }));
    const snapshot = felaSnapshot(<TableRow><TableCell /></TableRow>);
    expect(snapshot).toMatchSnapshot();
  });
});

describe('createTableRow', () => {
  it('should compose with styles overrides', () => {
    const TableRow = createTableRow(({ theme }) => ({
      verticalAlign: 'right',
      [`@media (min-width: ${theme.breakpoints.desktopLarge})`]: {
        width: '1000px'
      }
    }));
    const snapshot = felaSnapshot(<TableRow><TableCell /></TableRow>);
    expect(snapshot).toMatchSnapshot();
  });

  it('should be able to render a div', () => {
    const TableRow = createTableRow(({ theme }) => ({
      verticalAlign: 'right',
      [`@media (min-width: ${theme.breakpoints.desktopLarge})`]: {
        width: '1000px'
      }
    }));
    const snapshot = felaSnapshot(
      <TableRow is="div"><TableCell is="div" /></TableRow>
    );
    expect(snapshot).toMatchSnapshot();
  });
});
