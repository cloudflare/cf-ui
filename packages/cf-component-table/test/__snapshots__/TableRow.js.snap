// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow should compose with styles overrides 1`] = `
Object {
  "component": <tr
    className="cf-t744or"
>
    <td
        className="cf-1glop85"
        colSpan={undefined}
        rowSpan={undefined}
    />
</tr>,
  "styles": ".cf-t744or {
  border-color: #dedede;
  vertical-align: right
}
.cf-1glop85 {
  border-top: 1px solid #dedede;
  line-height: 1.5;
  padding: 0.73333333333rem;
  vertical-align: middle;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis
}
@media (max-width: 47.2em) {
  .cf-t744or {
    display: block;
    width: 100%;
    box-sizing: border-box
  }
  .cf-1glop85 {
    display: block;
    width: 100%;
    box-sizing: border-box;
    border-top-width: 0px
  }
}
@media screen and (max-width: 47.2em) {
  .cf-t744or:after {
    content: \\"\\";
    display: table;
    clear: both
  }
  .cf-t744or {
    display: block;
    padding: 0.73333333333rem;
    border-top: 1px solid #dedede;
    background-color: #ccc
  }
  .cf-1glop85 {
    display: block;
    border: none;
    float: left;
    clear: left;
    padding: 0px
  }
}
@media (min-width: 97.6em) {
  .cf-t744or {
    width: 1000px
  }
}",
}
`;

exports[`TableRow should render 1`] = `
Object {
  "component": <tr
    className="cf-1k2myrb"
>
    TableRow
</tr>,
  "styles": ".cf-1k2myrb {
  border-color: #dedede
}
@media (max-width: 47.2em) {
  .cf-1k2myrb {
    display: block;
    width: 100%;
    box-sizing: border-box
  }
}
@media screen and (max-width: 47.2em) {
  .cf-1k2myrb:after {
    content: \\"\\";
    display: table;
    clear: both
  }
  .cf-1k2myrb {
    display: block;
    padding: 0.73333333333rem;
    border-top: 1px solid #dedede;
    background-color: #ccc
  }
}",
}
`;

exports[`TableRow should render with type 1`] = `
Object {
  "component": <tr
    className="cf-7q8ucp"
>
    TableRow
</tr>,
  "styles": ".cf-7q8ucp {
  border-color: #d93c3e;
  background-color: #dd5153
}
@media (max-width: 47.2em) {
  .cf-7q8ucp {
    display: block;
    width: 100%;
    box-sizing: border-box
  }
}
@media screen and (max-width: 47.2em) {
  .cf-7q8ucp:after {
    content: \\"\\";
    display: table;
    clear: both
  }
  .cf-7q8ucp {
    display: block;
    padding: 0.73333333333rem;
    border-top: 1px solid #dedede;
    background-color: #ccc
  }
}",
}
`;

exports[`createTableRow should be able to render a div 1`] = `
Object {
  "component": <div
    bare={undefined}
    bordered={undefined}
    className="cf-1edt7t"
    condensed={undefined}
    hover={undefined}
    is="div"
    rowIndex={undefined}
    striped={undefined}
    tbodyIndex={undefined}
    tfootIndex={undefined}
    theadIndex={undefined}
    type={undefined}
>
    <div
        align={undefined}
        bare={undefined}
        bordered={undefined}
        cellIndex={undefined}
        className="cf-kk9pov"
        colSpan={undefined}
        condensed={undefined}
        headers={undefined}
        hover={undefined}
        is="div"
        rowIndex={undefined}
        rowSpan={undefined}
        rowType={undefined}
        striped={undefined}
        tbodyIndex={undefined}
        tfootIndex={undefined}
        theadIndex={undefined}
        width={undefined}
    />
</div>,
  "styles": ".cf-1edt7t {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-wrap: nowrap;
  -webkit-box-lines: nowrap;
  -webkit-flex-wrap: nowrap;
  justify-content: space-around;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-around;
  align-items: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  border-color: #dedede;
  vertical-align: right
}
.cf-kk9pov {
  display: inline-block;
  flex: auto;
  -webkit-flex: auto;
  border-top: 1px solid #dedede;
  line-height: 1.5;
  padding: 0.73333333333rem;
  vertical-align: middle;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  border-color: #dedede
}
@media (max-width: 47.2em) {
  .cf-1edt7t {
    display: block;
    width: 100%;
    box-sizing: border-box
  }
  .cf-kk9pov {
    display: block;
    width: 100%;
    box-sizing: border-box;
    border-top-width: 0px
  }
}
@media screen and (max-width: 47.2em) {
  .cf-1edt7t:after {
    content: \\"\\";
    display: table;
    clear: both
  }
  .cf-1edt7t {
    display: block;
    padding: 0.73333333333rem;
    border-top: 1px solid #dedede;
    background-color: #ccc
  }
  .cf-kk9pov {
    display: block;
    border: none;
    float: left;
    clear: left;
    padding: 0px
  }
}
@media (min-width: 97.6em) {
  .cf-1edt7t {
    width: 1000px
  }
}",
}
`;

exports[`createTableRow should compose with styles overrides 1`] = `
Object {
  "component": <tr
    className="cf-t744or"
>
    <td
        className="cf-1glop85"
        colSpan={undefined}
        rowSpan={undefined}
    />
</tr>,
  "styles": ".cf-t744or {
  border-color: #dedede;
  vertical-align: right
}
.cf-1glop85 {
  border-top: 1px solid #dedede;
  line-height: 1.5;
  padding: 0.73333333333rem;
  vertical-align: middle;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis
}
@media (max-width: 47.2em) {
  .cf-t744or {
    display: block;
    width: 100%;
    box-sizing: border-box
  }
  .cf-1glop85 {
    display: block;
    width: 100%;
    box-sizing: border-box;
    border-top-width: 0px
  }
}
@media screen and (max-width: 47.2em) {
  .cf-t744or:after {
    content: \\"\\";
    display: table;
    clear: both
  }
  .cf-t744or {
    display: block;
    padding: 0.73333333333rem;
    border-top: 1px solid #dedede;
    background-color: #ccc
  }
  .cf-1glop85 {
    display: block;
    border: none;
    float: left;
    clear: left;
    padding: 0px
  }
}
@media (min-width: 97.6em) {
  .cf-t744or {
    width: 1000px
  }
}",
}
`;
